### PR TITLE
[fix bug 1380490] L10n/JS updates for Sync page.

### PR DESCRIPTION
**Pull request by jpetto**
## Description

- Cleans up old L10n tags that are no longer needed (confirmed w/Peying)
- Integrates `sync-en.html` template into `sync.html` & wraps previously en only strings
- Big JS refactor for easier unit testing
- Accounts for Fx iOS users (same messaging as Fx Android)

## Issue / Bugzilla link

https://bugzilla.mozilla.org/show_bug.cgi?id=1380490

## Testing

Ensure no regressions. Sanity check JS refactoring.

### CODE REVIEW

File: media/js/firefox/features-sync.js [hunk 1]
```diff
@@ -1,104 +1,126 @@
...
-
-                // Query if the UITour API is working before we use the API
-                Mozilla.UITour.getConfiguration('sync', function (config) {
-
-                    // Variation #1: Firefox 31+ signed IN to Sync (default)
-                    if (config.setup) {
-                        swapState('state-fx-31-signed-in');
-                        state = 'Firefox 31 or Higher: Signed-In';
+    SyncPage.setBodyClass = function(stateClass) {
+        var classes = SyncPage.body.getAttribute('class');
```
Comment (alexgibson) [inline media/js/firefox/features-sync.js]: I don't think this works in older browsers like IE, should be `SyncPage.body.className;`?

File: media/js/firefox/features-sync.js [hunk 2]
```diff
@@ -1,104 +1,126 @@
...
+        classes = classes.replace('state-default', stateClass);
+        SyncPage.body.setAttribute('class', classes);
```
Comment (alexgibson) [inline media/js/firefox/features-sync.js]: Again should this be `SyncPage.body.className = classes;`?

File: media/js/firefox/features-sync.js [hunk 3]
```diff
@@ -1,104 +1,126 @@
...
+                'event': 'sync-click',
+                'browser': SyncPage.state
+            });
+
+            Mozilla.UITour.showFirefoxAccounts(SyncPage.params.utmParamsFxA());
+        };
+
+        // Firefox Sync sign in flow button
+        // Only displayed for Fx 31+
+        SyncPage.ctaSync.addEventListener('click', SyncPage.ctaSyncClick);
```
Comment (alexgibson) [inline media/js/firefox/features-sync.js]: If this button is shown only for Fx 31+ should we only bind the event if that conditional is true?

File: bedrock/settings/static_media.py [hunk 4]
```diff
@@ -1365,6 +1365,7 @@
         'source_filenames': (
             'js/base/uitour-lib.js',
             'js/firefox/features-sync.js',
+            'js/firefox/features-sync-init.js',
```
Comment (jpetto) [inline bedrock/settings/static_media.py]: Yeah, good call. Will do.

File: media/js/firefox/features-sync.js (hunk 1 above)
Comment (jpetto) [inline media/js/firefox/features-sync.js]: `getAttribute` and `setAttribute` work back to IE 6, so I think we're good. 😉

File: media/js/firefox/features-sync.js (hunk 3 above)
Comment (jpetto) [inline media/js/firefox/features-sync.js]: Good call.

File: media/js/firefox/features-sync.js (hunk 1 above)
Comment (alexgibson) [inline media/js/firefox/features-sync.js]: Not for `body` (try it!)

Comment (alexgibson) [inline media/js/firefox/features-sync.js]: To add more clarity, 'class' is not a valid attribute. As per the spec, it's actually a property `className`. Modern browsers support it, but not every browser

Comment (jpetto) [inline media/js/firefox/features-sync.js]: Oh, wow. Weak!

(Since my IE 6 VM reached EOL, I don't have way to test there...)

**Pull request merged**